### PR TITLE
chore: replacing KafkaRealm with set provider

### PR DESCRIPTION
**Pull request by akoserwal**
Updating all required occurrences of KafkaRealm with GetRealmProvider.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has created for changes required on the client side

### CODE REVIEW

File: internal/connector/internal/routes/route_loader.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func (s *options) AddRoutes(mainRouter *mux.Router) error {
 		agentRouter.HandleFunc("/namespaces", s.ConnectorClusterHandler.GetAgentNamespaces).Methods(http.MethodGet)
 		agentRouter.HandleFunc("/namespaces/{namespace_id}", s.ConnectorClusterHandler.GetNamespace).Methods(http.MethodGet)
 		agentRouter.HandleFunc("/deployments/{deployment_id}/status", s.ConnectorClusterHandler.UpdateDeploymentStatus).Methods(http.MethodPut)
-		auth.UseOperatorAuthorisationMiddleware(agentRouter, s.KeycloakService.GetConfig().KafkaRealm.ValidIssuerURI, "connector_cluster_id", s.AuthAgentService)
+		auth.UseOperatorAuthorisationMiddleware(agentRouter, s.KeycloakService.GetConfig().GetSSOProviderRealm().ValidIssuerURI, "connector_cluster_id", s.AuthAgentService)
```
Comment (machi1990) [inline internal/connector/internal/routes/route_loader.go:146]:
```suggestion
		auth.UseOperatorAuthorisationMiddleware(agentRouter, s.KeycloakService.GetRealmConfig().ValidIssuerURI, "connector_cluster_id", s.AuthAgentService)
```

File: internal/kafka/internal/routes/route_loader.go [hunk 2]
```diff
@@ -149,7 +149,7 @@ func (s *options) buildApiBaseRouter(mainRouter *mux.Router, basePath string, op
 	apiV1MetricsFederateRouter.HandleFunc("", metricsHandler.FederateMetrics).
 		Name(logger.NewLogEvent("get-federate-metrics", "get federate metrics by id").ToString()).
 		Methods(http.MethodGet)
-	apiV1MetricsFederateRouter.Use(auth.NewRequireIssuerMiddleware().RequireIssuer([]string{s.ServerConfig.TokenIssuerURL, s.Keycloak.GetConfig().KafkaRealm.ValidIssuerURI}, errors.ErrorUnauthenticated))
+	apiV1MetricsFederateRouter.Use(auth.NewRequireIssuerMiddleware().RequireIssuer([]string{s.ServerConfig.TokenIssuerURL, s.Keycloak.GetConfig().GetSSOProviderRealm().ValidIssuerURI}, errors.ErrorUnauthenticated))
```
Comment (machi1990) [inline internal/kafka/internal/routes/route_loader.go:152]:
```suggestion
	apiV1MetricsFederateRouter.Use(auth.NewRequireIssuerMiddleware().RequireIssuer([]string{s.ServerConfig.TokenIssuerURL, s.Keycloak.GetRealmConfig().ValidIssuerURI}, errors.ErrorUnauthenticated))
```

File: internal/kafka/internal/routes/route_loader.go [hunk 3]
```diff
@@ -233,7 +233,7 @@ func (s *options) buildApiBaseRouter(mainRouter *mux.Router, basePath string, op
 		Name(logger.NewLogEvent("list-dataplane-kafkas", "list all dataplane kafkas").ToString()).
 		Methods(http.MethodGet)
 	// deliberately returns 404 here if the request doesn't have the required role, so that it will appear as if the endpoint doesn't exist
-	auth.UseOperatorAuthorisationMiddleware(apiV1DataPlaneRequestsRouter, s.Keycloak.GetConfig().KafkaRealm.ValidIssuerURI, "id", s.ClusterService)
+	auth.UseOperatorAuthorisationMiddleware(apiV1DataPlaneRequestsRouter, s.Keycloak.GetConfig().GetSSOProviderRealm().ValidIssuerURI, "id", s.ClusterService)
```
Comment (machi1990) [inline internal/kafka/internal/routes/route_loader.go:236]:
```suggestion
	auth.UseOperatorAuthorisationMiddleware(apiV1DataPlaneRequestsRouter, s.Keycloak.GetRealmConfig().ValidIssuerURI, "id", s.ClusterService)
```

**Review by machi1990 — CHANGES_REQUESTED** [review]
left a couple of suggestions since we can use the `.GetRealmConfig()` method from the keycloak service.
Otherwise, it lgtm.

**Comment by akoserwal** [comment]
@dhirajsb @valdar

File: pkg/client/keycloak/config.go [hunk 4]
```diff
@@ -50,6 +50,18 @@ type KeycloakRealmConfig struct {
...
+	switch provider {
+	case MAS_SSO:
+		return kc.KafkaRealm
+	case REDHAT_SSO:
+		return kc.RedhatSSORealm
+	default:
+		return kc.KafkaRealm
+	}
+}
+
```
Comment (ziccardi) [inline pkg/client/keycloak/config.go:64]: By convention, Getters in Go should not have the `Get` prefix (see https://go.dev/doc/effective_go#Getters)
```suggestion
func (kc *KeycloakConfig) SSOProviderRealm() *KeycloakRealmConfig {
	provider := kc.SelectSSOProvider
	switch provider {
	case MAS_SSO:
		return kc.KafkaRealm
	case REDHAT_SSO:
		return kc.RedhatSSORealm
	default:
		return kc.KafkaRealm
	}
}

```

**Review by ziccardi — APPROVED** [review]
Just a minor comment. LGTM

**Pull request merged**